python/grass/jupyter/utils.py

```diff
@@ -0,0 +1,100 @@
+#
+# AUTHOR(S): Caitlin Haedrich <caitlin DOT haedrich AT gmail>
+#
+# PURPOSE:   This module contains utility functions for InteractiveMap.
+#
+# COPYRIGHT: (C) 2021 Caitlin Haedrich, and by the GRASS Development Team
+#
+#            This program is free software under the GNU Gernal Public
+#            License (>=v2). Read teh file COPYING that comes with GRASS
+#            for details.
+
+import os
+import grass.script as gs
+
+
+def convert_coordinates_to_latlon(x, y, proj_in):
+    """This function reprojects coordinates to WGS84, the required
+    projection for vectors in folium.
+
+    Arguments:
+        x -- x coordinate (string)
+        y -- y coordinate (string)
+        proj_in -- proj4 string of location (for example, the output
+        of g.region run with the `g` flag."""
+
+    # Reformat input
+    coordinates = f"{x}, {y}"
+    # Reproject coordinates
+    new_coords = gs.read_command(
+        "m.proj",
+        coordinates=coordinates,
+        proj_in=proj_in,
+        separator="comma",
+        flags="do",
+    )
+    # Reformat from string to array
+    new_coords = new_coords.strip()  # Remove '\n' at end of string
+    new_coords = new_coords.split(",")  # Split on comma
+    new_coords = [float(value) for value in new_coords]  # Convert to floats
+    return new_coords[1], new_coords[0]  # Return Lat and Lon
+
+
+def get_region(env=None):
+    """Returns current computational region as dictionary.
+    Adds long key names.
+    """
+    region = gs.region(env=env)
+    region["east"] = region["e"]
+    region["west"] = region["w"]
+    region["north"] = region["n"]
+    region["south"] = region["s"]
+    return region
+
+
+def get_location_proj_string(env=None):
+    out = gs.read_command("g.proj", flags="jf", env=env)
+    return out.strip()
+
+
+def reproject_region(region, from_proj, to_proj):
```

Comment:
Ideally, you would have a third function which would have the m.proj call and surrounding string manipulation and that may avoid the need for two calls of `convert_coordinates_to_latlon` which are good API-wise, but not ideal performance-wise, but it might be too cumbersome in practice.

Reply:
I managed to avoid calling `convert_coordinates_to_latlong` and deleted it. Do you think it's still necessary to have a third function here?

Reply:
No, with the current code, this work well.